engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -154,6 +155,34 @@ namespace dmGameSystem
         info->m_RefCount++;
     }
 
+    static void CopyBufferToVertexBuffer(dmBuffer::HBuffer buffer, dmGraphics::HVertexBuffer vertex_buffer, uint32_t vert_size, uint32_t elem_count, dmGraphics::BufferUsage buffer_usage)
+    {
+        uint8_t* bytes = 0x0;
+        uint32_t size = 0;
+        dmBuffer::Result r = dmBuffer::GetBytes(buffer, (void**)&bytes, &size);
+        assert(r == dmBuffer::RESULT_OK);
+
+        dmGraphics::SetVertexBufferData(vertex_buffer, vert_size * elem_count, bytes, buffer_usage);
+    }
+
+    static void CreateVertexBuffer(MeshWorld* world, dmGameSystem::BufferResource* br, uint32_t vert_size)
+    {
+        dmGraphics::HVertexBuffer vertex_buffer = GetVertexBuffer(world, br->m_NameHash);
+        if (!vertex_buffer)
+        {
+            vertex_buffer = AllocVertexBuffer(world, world->m_GraphicsContext);
+            AddVertexBufferInfo(world, br->m_NameHash, vertex_buffer, br->m_Version); // ref count == 1
+
+            uint32_t elem_count = br->m_ElementCount;
+
+            CopyBufferToVertexBuffer(br->m_Buffer, vertex_buffer, vert_size, elem_count, dmGraphics::BUFFER_USAGE_STREAM_DRAW);
+        }
+        else
+        {
+            IncRefVertexBuffer(world, br->m_NameHash);
+        }
+    }
```

Comment:
A reusable function for when the component is created, as well as when a new `.buffer` is assigned.